tensorflow/python/eager/forwardprop_test.py

```diff
@@ -90,6 +90,28 @@ def _jacfwd(f, primals):
   return nest.pack_sequence_as(primals, jac_flat)
 
 
+def _jvp_batch(f, primal, tangents):
+
+  return control_flow_ops.vectorized_map(
```

Comment:
You may want to note that this traces `f` and the rest of the computation every time it runs. I think the result is what we'll eventually want, but we'll want `f` to run eagerly.

Reply:
Fixed that. But I'm not sure if retracing in `_jacfwd` needs to be handled and if yes then how. I'm using `_jvp_batch_matmul` just to check correctness for now.

Reply:
Every time _jvp_batch runs, it creates a new def_function.function object, which will start with no traces and have to build one the first time it's called. That's what vectorized_map was doing itself I think. But it's self-documenting now, so maybe that works instead of a note/comment.

I don't think we should try fixing retracing for this PR. Fine to just add the tests. This API issue (no vectorized tangents executing eagerly) is one of the main reasons we need your summer project.